server/gitrest/packages/gitrest-base/src/utils/redisFs/redisFsManager.ts

```diff
@@ -0,0 +1,457 @@
+			void this.redisFs.delete(key);
+		});
+	}
+
+	/**
+	 * Returns an fs.Stat object for a given filepath. fs.Stat provides information about
+	 * a file or folder.
+	 * `fs.Stats` would usually include information like file size and modified time. However, it is
+	 * very expensive to fetch that information from redis every time. Because we don't employ
+	 * fields like `stats.size` and `stats.mtime` in our specific use-case scenario of isomorphic-git,
+	 * we can avoid making "get properties" requests to Az storage, as the only other types of information
+	 * we need are whether a path exists and whether it refers to a file or directory. We can compute that
+	 * independently. (Please note that we are initially enabling this change for only a subset of tenants)
+	 * For more info on how isomorphic-git uses this, see:
+	 * https://github.com/isomorphic-git/isomorphic-git/blob/main/src/models/FileSystem.js#L61
+	 * https://github.com/isomorphic-git/isomorphic-git/blob/main/src/models/FileSystem.js#L191
+	 * Neither isomorphic-git nor GitRest currently use the options parameter.
+	 */
+	public async stat(
+		filepath: PathLike,
+		options?: StatOptions & { bigint?: false | undefined },
+	): Promise<Stats>;
+	public async stat(
+		filepath: PathLike,
+		options: StatOptions & { bigint: true },
+	): Promise<BigIntStats>;
+	public async stat(filepath: PathLike, options?: StatOptions): Promise<Stats | BigIntStats>;
+	public async stat(filepath: PathLike, options?: any): Promise<Stats | BigIntStats> {
+		const filepathString = filepath.toString();
+		const data = await executeRedisFsApi(
+			async () => this.redisFs.get<Buffer | string>(filepathString),
+			RedisFsApis.Stat,
+			RedisFSConstants.RedisFsApi,
+			this.redisFsConfig.enableRedisFsMetrics,
+			this.redisFsConfig.redisApiMetricsSamplingPeriod,
+			{
+				filepathString,
+			},
+			true,
+		);
+
+		if (!data) {
+			throw new RedisFsError(SystemErrors.ENOENT, filepath.toString());
+		}
+		return getStats();
+	}
+
+	/**
+	 * Asynchronously removes files and directories (modeled on the standard POSIX `rm` utility).
+	 * Even though `rm` is optional to isomorphic-git, we need to implement it because of this:
+	 * https://github.com/isomorphic-git/isomorphic-git/blob/main/src/models/FileSystem.js#L19-L24.
+	 * isomorphic-git provides a workaround implementation for `rm`, but it is not used if our `rmdir`
+	 * implementation takes more than 1 argument. And since we implement `rmdir` taking 2 arguments
+	 * in order to conform with `IFileSystemPromises`, we end up having to provide an `rm` implementation.
+	 * For more info on how isomorphic-git uses this, see:
+	 * https://github.com/isomorphic-git/isomorphic-git/blob/main/src/models/FileSystem.js#L155
+	 */
+	public async rm(filepath: PathLike, options?: RmOptions): Promise<void> {
+		const filepathString = filepath.toString();
+		await executeRedisFsApi(
+			async () => this.redisFs.delete(filepath.toString()),
```

Comment:
If this is a key prefix in Redis, does it remove all keys with that prefix? Or does it fail?

Reply:
Or maybe that does not matter because `mkdir` will create keys with empty values to represent directories?

What if we have the following situation:
1. mkdir used for `folderA/folderB` and recursive is set
2. mkdir ignores recursive so it will create a single key, `folderA/folderB`
3. rm is called for `folderA`

Would we get an error?

Reply:
Correct..in that case we would get an error. But ideally we should not run into that situation. And even if we do, getting an error that does not affect us seems like a better compromise than recursively creating all the paths and adding a lot more keys just to avoid this error